jumanji/environments/routing/connector/generator.py

```diff
@@ -293,272 +299,91 @@ def _initialize_starts_and_first_move(
-
-        Args:
-            grid: the current layout of the board i.e. current grid.
-            cell: the flat index of the cell to find adjacent cells of.
-
-        Returns:
-            A chex.Array of size 4 with the flat indices of adjacent cells.
-        """
-        adjacent_cells = self._adjacent_cells(cell)
-        # Get the wire id of the current cell
-        value = grid[jnp.divmod(cell, self.grid_size)]
-        wire_id = (value - 1) // 3
-
-        available_cells_mask = jax.vmap(self._is_cell_free, in_axes=(None, 0))(grid, adjacent_cells)
-        # Also want to check if the cell is touching itself more than once
-        touching_cells_mask = jax.vmap(self._is_cell_doubling_back, in_axes=(None, None, 0))(
-            grid, wire_id, adjacent_cells
-        )
-        available_cells_mask = available_cells_mask & touching_cells_mask
-        available_cells = jnp.where(available_cells_mask, adjacent_cells, -1)
-        return available_cells
-
-    def _is_cell_free(
-        self,
-        grid: chex.Array,
-        cell: chex.Array,
-    ) -> chex.Array:
-        """Check if a given cell is free, i.e. has a value of 0.
-
-        Args:
-            grid: the current grid of the board.
-            cell: the flat index of the cell to check.
-
-        Returns:
-            Boolean indicating whether the cell is free or not.
-        """
-        coordinate = jnp.divmod(cell, self.grid_size)
-        return (cell != -1) & (grid[coordinate] == 0)
+        return action
 
-    def _is_cell_doubling_back(
-        self,
-        grid: chex.Array,
-        wire_id: int,
-        cell: int,
+    @staticmethod
+    def _calculate_action_probs(
+        current_position: chex.Array,
+        start_position: chex.Array,
+        action_mask: chex.Array,
+        temperature: float,
     ) -> chex.Array:
-        """Checks if moving into an adjacent position results in a wire doubling back on itself.
-
-        Check if the cell is touching any of the wire's own cells more than once.
-        This means looking for surrounding cells of value 3 * wire_id + POSITION or
-        3 * wire_id + PATH.
-        """
-        # Get the adjacent cells of the current cell
-        adjacent_cells = self._adjacent_cells(cell)
-
-        def is_cell_doubling_back_inner(grid: chex.Array, cell: chex.Array) -> chex.Array:
-            coordinate = jnp.divmod(cell, self.grid_size)
-            cell_value = grid[tuple(coordinate)]
-            touching_self = (
-                (cell_value == 3 * wire_id + POSITION)
-                | (cell_value == 3 * wire_id + PATH)
-                | (cell_value == 3 * wire_id + TARGET)
-            )
-            return (cell != -1) & touching_self
-
-        # Count the number of adjacent cells with the same wire id
-        doubling_back_mask = jax.vmap(is_cell_doubling_back_inner, in_axes=(None, 0))(
-            grid, adjacent_cells
+        displacement = current_position - start_position
```

Comment:
Can you add some docs/comments in this one to explain how and why we're doing this